src/app/containers/MostRead/Canonical/index.jsx

```diff
@@ -1,45 +1,119 @@
 import React, { useEffect, useState } from 'react';
 import 'isomorphic-fetch';
-import { string } from 'prop-types';
+import { number, oneOf, string, shape } from 'prop-types';
+import { scriptPropType } from '@bbc/gel-foundations/prop-types';
+import Timestamp from '@bbc/psammead-timestamp-container';
+import MostRead from '@bbc/psammead-most-read';
+import mostReadTranslationsPropTypes from '#models/propTypes/mostRead';
 import webLogger from '#lib/logger.web';
+import { shouldRenderLastUpdated, mostReadRecordIsFresh } from '../utilities';
 
 const logger = webLogger();
 
-const CanonicalMostRead = ({ endpoint }) => {
-  const [promos, setPromo] = useState([]);
-  const [data, setData] = useState({});
+const lastUpdated = ({ locale, prefix, script, service, timestamp }) => (
+  <Timestamp
+    timestamp={timestamp}
+    dateTimeFormat="YYYY-MM-DD"
+    prefix={prefix}
+    format="LL"
+    script={script}
+    service={service}
+    locale={locale}
+  />
+);
 
-  const handleResponse = async response => {
-    const mostReadData = await response.json();
-    setPromo(mostReadData.records.slice(0, 10));
-    setData(mostReadData);
-  };
+lastUpdated.propTypes = {
+  locale: string.isRequired,
+  prefix: string.isRequired,
+  script: shape(scriptPropType).isRequired,
+  service: string.isRequired,
+  timestamp: number.isRequired,
+};
+
+const CanonicalMostRead = ({
+  dir,
+  endpoint,
+  translations,
+  script,
+  service,
+  locale,
+}) => {
+  const [items, setItems] = useState([]);
 
   useEffect(() => {
+    const handleResponse = async response => {
+      const mostReadData = await response.json();
+
+      // do not show most read if lastRecordUpdated is greated than 35min
+      if (!mostReadRecordIsFresh(mostReadData.lastRecordTimeStamp)) {
+        return;
+      }
+
+      // extracting the data we need from the api
+      const sortedItems = [];
+
+      mostReadData.records
+        .slice(0, 10)
+        .forEach(({ id, promo: { headlines, timestamp, locators } }) => {
+          const renderTimestampContainer = shouldRenderLastUpdated(
+            timestamp,
+          ) ? (
+            lastUpdated({
+              prefix: translations.lastUpdated,
+              script,
+              service,
+              timestamp,
+              locale,
+            })
+          ) : (
+            <></>
```

Comment:
https://github.com/bbc/psammead/blob/latest/packages/components/psammead-most-read/src/testHelpers/itemsHelper.jsx#L37

I made this a required prop in psammead 🤦‍♂ , it can be not required so that i can just pass null here

Reply:
@thekp I don't think [timespamp](https://github.com/bbc/psammead/blob/463a3d406abba89b7527f5eb8a67306a17b67e17/packages/components/psammead-most-read/src/testHelpers/itemsHelper.jsx#L37) need to be a  required prop.

Reply:
https://github.com/bbc/psammead/blob/463a3d406abba89b7527f5eb8a67306a17b67e17/packages/components/psammead-most-read/src/testHelpers/itemsHelper.jsx#L37

Reply:
yup agreed

Reply:
https://github.com/bbc/psammead/pull/2870

fixed here